### PR TITLE
zephyr: zephyr_reporter: modify cycleName rule

### PR DESCRIPTION
**Pull request by lynnlincbn**
Base on commit REMOVED PRIVATE URL to modify cycleName rule

Signed-off-by: lynn <lynn_lin@compalbn.com>

---
This change is [Reviewable](https://reviewable.io/reviews/lgirdk/boardfarm/261)

### CODE REVIEW

**Comment by mattsm** [comment]
I see this, and understand but need to sit down and take a closer look and test.. will take time to review.

**Comment by lynnlincbn** [comment]
Hi Matthew, did you have any update for your review? thanks.

File: zephyr/zephyr_reporter.py [hunk 1]
```diff
@@ -119,7 +119,7 @@ def update_zephyr(test_cases_list):
         proj = jira.project(z["project"])
         verid = get_jira_release_id(z['release'], jira, proj)
         cycleName = z["cycle"]
-        cycleName = cycleName + "_" + str((datetime.datetime.now()).strftime("%Y%m%d%H%M%S"))
+        cycleName = cycleName.replace("date", str((datetime.datetime.now()).strftime("%Y%m%d")))
```
Comment (mattsm) [inline zephyr/zephyr_reporter.py]: In the future I think we just need to change the config file from JSON to a python file and we can encode whatever logic we want in python for each config we use...

For example

```
import datetime

config = {
	"report": "output.xml",
	"metafile": "zephyr/boardfarm_tc_meta_file.csv",
	"project": "PROJECT",
	"release": "RELEASE",
	"environment": "Test",
	"cycle": "CYCLE_%s_ABC" % datetime.datetime.now()).strftime("%Y%m%d%H%M%S"),
	"build": "DemoBuild",
	"jira_url": "JIRA_URL",
	"user": "USER",
	"passwd": "PASS",
	"updateautomationstatus": true,
	"status_codes": {
		"UNEXECUTED": -1,
		"PASS": 1,
		"FAIL": 2,
		"NOT TESTED": 7
	}
}
```

Comment (mattsm) [reply zephyr/zephyr_reporter.py]: If a change is required can you change the date replace from 'date' to '<DATE>'

File: zephyr/zephyr_reporter.py [hunk 2]
```diff
@@ -136,46 +136,47 @@ def update_zephyr(test_cases_list):
...
+                     exec_id,
+                     log_data,
+                     status_code_dict=z['status_codes'])
+                else:
+                    ret = reporter.set_execution(result,
+                     exec_id,
+                     log_data)
+
+                if ret.status_code != requests.codes.ok:
+                    raise Exception("Error = %s, when trying to set execution status" % ret)
```
Comment (mattsm) [inline zephyr/zephyr_reporter.py]: Is this really needed when the other change merged? 

https://github.com/lgirdk/boardfarm/commit/36b8fac7872ea11d9b2495451f360c83b1327cfd

File: zephyr/zephyr_reporter.py (hunk 1 above)
Comment (mattsm) [reply zephyr/zephyr_reporter.py]:
```
<DATE>
```

File: zephyr/zephyr_reporter.py (hunk 2 above)
Comment (lynnlincbn) [reply zephyr/zephyr_reporter.py]: I don't know what other change merged? Can you tell more detail for this?
This commit is to fix if test_cases_list[i] ==[] will error on line 139

File: zephyr/zephyr_reporter.py (hunk 1 above)
Comment (lynnlincbn) [reply zephyr/zephyr_reporter.py]: Not only date need to modify, also "release" and "cycle" need to change.
But for now I modify those manually.
Can you help me how can I pass value from boardfarm to zephyr_reporter?
I can snmpget fw version and modify to release and cycle format, but can't pass those parameters now.

Comment (mattsm) [reply zephyr/zephyr_reporter.py]: We discusses this privately. Please use the info_for_remote dict and self.logged in the tests..

File: zephyr/zephyr_reporter.py (hunk 2 above)
Comment (mattsm) [reply zephyr/zephyr_reporter.py]: I think you just need to fix the for loop:

for test_case in test_cases_list:

should be 

for i in range(len(test_cases_list)):

**Comment by mattsm** [comment]
Does not seem to be interest in fixing this, closing for now.

**Pull request closed without merging**